### DISCUSSION TITLE
2022-03-14 weekly planning

### DISCUSSION THREAD

**Discussion by kgodey** [mathesar-foundation/mathesar#1162] [Maintainer discussions]
## About
At the start of each week, we start a discussion covering Mathesar's progress, team member check ins, and general updates. Everyone is welcome to add a new topic to discuss or to comment on existing topics.

## Discussion topics
- [Last Week in Mathesar](https://github.com/centerofci/mathesar/discussions/1162#discussioncomment-2343236)
- [Frontend priorities](https://github.com/centerofci/mathesar/discussions/1162#discussioncomment-2343237)
- [Backend priorities](https://github.com/centerofci/mathesar/discussions/1162#discussioncomment-2343239)
- [Design priorities](https://github.com/centerofci/mathesar/discussions/1162#discussioncomment-2343241)
- [Product priorities](https://github.com/centerofci/mathesar/discussions/1162#discussioncomment-2343242)
- [Comms Assignee schedule](https://github.com/centerofci/mathesar/discussions/1162#discussioncomment-2343243)
- [Sean's check in](https://github.com/centerofci/mathesar/discussions/1162#discussioncomment-2357806)
- [Dominykas' check in](https://github.com/centerofci/mathesar/discussions/1162#discussioncomment-2358032)
- [Brent's check in](https://github.com/centerofci/mathesar/discussions/1162#discussioncomment-2359113)
- [Kriti's check in](https://github.com/centerofci/mathesar/discussions/1162#discussioncomment-2359122)
- [Pavish's check in](https://github.com/centerofci/mathesar/discussions/1162#discussioncomment-2359404)
- [Ghislaine's check in](https://github.com/centerofci/mathesar/discussions/1162#discussioncomment-2360945)
- [Mukesh's check in](https://github.com/centerofci/mathesar/discussions/1162#discussioncomment-2362777)

## Weekly check-in
@centerofci/mathesar-core please post your weekly check-in as a top-level response and edit the discussion topics above to add your check in as a topic. Others are also welcome to respond.

Prompts:
- How did your most recent work go?
- What are you planning to work on soon?
- Do you anticipate any dependencies or blockers that someone else on the team can help you with?

If you'd like to add anything else to your check-in that's not related to the prompts, feel free to do so.

**Comment by kgodey** [comment]
## Last Week in Mathesar

### News
Mathesar was [accepted as a mentoring organization for Google Summer of Code 2022](https://centerofci.org/2022/03/08/mathesar-participating-in-google-summer-of-code-2022/)! We've already welcomed a few new contributors to the repo and look forward to the program.

### Feedback Requested
- https://github.com/centerofci/mathesar/discussions/1154
- https://github.com/centerofci/mathesar/discussions/1150

### Complete
- New (temporary) logo for Mathesar - see [our README](https://github.com/centerofci/mathesar#readme)
- **Discussions**:
    - https://github.com/centerofci/mathesar/discussions/1147
    - https://github.com/centerofci/mathesar/discussions/1136
    - https://github.com/centerofci/mathesar/discussions/1034
    - https://github.com/centerofci/mathesar/discussions/1015
- **Full stack and general work**:
    - https://github.com/centerofci/mathesar/issues/896
- **Backend work**:
    - https://github.com/centerofci/mathesar/issues/1094
    - https://github.com/centerofci/mathesar/issues/431
    - https://github.com/centerofci/mathesar/issues/960
    - https://github.com/centerofci/mathesar/issues/994
    - https://github.com/centerofci/mathesar/issues/1146
- **Frontend work**:
    - https://github.com/centerofci/mathesar/issues/742
    - https://github.com/centerofci/mathesar/issues/1074
    - https://github.com/centerofci/mathesar/pull/1133
- **Documentation work**:
    - https://github.com/centerofci/mathesar-wiki/pull/37

**Comment by kgodey** [comment]
## Frontend priorities

1. Get open PRs merged:
    1. https://github.com/centerofci/mathesar/pull/1144
    1. https://github.com/centerofci/mathesar/pull/1155
    1. https://github.com/centerofci/mathesar/pull/1159
    1. https://github.com/centerofci/mathesar/pull/1160
2. Make progress on high-priority issues for [Initial Data Types](https://github.com/centerofci/mathesar/milestone/11) milestone:
    1. #970
    1. #930 
    1. #1108
    1. #257 
    1. #775
3. Continue working on other [Initial Data Types](https://github.com/centerofci/mathesar/milestone/11) frontend issues.

@pavish @seancolsen Please comment below with thoughts/concerns.

**Reply by seancolsen** [reply]
Looks good

**Reply by pavish** [reply]
Sounds good to me

**Comment by kgodey** [comment]
## Backend priorities

The frontend is catching up fast to the backend, so we really need to get backend work for [Initial Data Types](https://github.com/centerofci/mathesar/milestone/11)completed and start on [Links between Tables](https://github.com/centerofci/mathesar/milestone/47) ASAP.

1. Get open PRs completed and merged:
    1. https://github.com/centerofci/mathesar/pull/1157
    1. https://github.com/centerofci/mathesar/pull/1103
    1. https://github.com/centerofci/mathesar/pull/1148
2. Complete remaining backend work for [Initial Data Types](https://github.com/centerofci/mathesar/milestone/11):
    1. @dmos62:
        1. https://github.com/centerofci/mathesar/issues/1100
        1. https://github.com/centerofci/mathesar/issues/1036
        1. https://github.com/centerofci/mathesar/issues/1037
    2. @mathemancer
        1. https://github.com/centerofci/mathesar/issues/845
        2. **Grouping**:
            1. https://github.com/centerofci/mathesar/issues/408 
            2. https://github.com/centerofci/mathesar/issues/411 
            3. https://github.com/centerofci/mathesar/issues/414
            4. https://github.com/centerofci/mathesar/issues/384
            5. https://github.com/centerofci/mathesar/issues/429
            6. https://github.com/centerofci/mathesar/issues/419
            7. https://github.com/centerofci/mathesar/issues/432
    3. @silentninja:
        1. https://github.com/centerofci/mathesar/issues/839
        1. https://github.com/centerofci/mathesar/issues/996
        1. https://github.com/centerofci/mathesar/issues/1105
    4. Not assigned to core team members:
        1. https://github.com/centerofci/mathesar/issues/1121 (unassigned)
        1. https://github.com/centerofci/mathesar/issues/1145 (assigned to @ManishShah120)
3. Start work on backend work for [Links between Tables](https://github.com/centerofci/mathesar/milestone/47), some ideas:
    1. https://github.com/centerofci/mathesar/issues/943
    2. https://github.com/centerofci/mathesar/issues/944
    3. https://github.com/centerofci/mathesar/issues/973
    4. https://github.com/centerofci/mathesar/issues/972
4. #1008

@dmos62 @mathemancer @silentninja Please comment below with thoughts/concerns.

**Reply by mathemancer** [reply]
I think this makes sense.

**Comment by kgodey** [comment]
## Design priorities

1. Publish spec and start review process for #1065
2. Continue iterating on #978 

@ghislaineguerin @ppii775 Please comment below with thoughts/concerns

**Reply by ghislaineguerin** [reply]
@kgodey looks good

**Comment by kgodey** [comment]
## Product priorities

_No product work is planned for this week._

**Comment by kgodey** [comment]
## Comms Assignee schedule

| Week | Person |
|-|-|
| 2022-03-14 to 2022-03-18 | Pavish |
| 2022-03-21 to 2022-03-25 | Mukesh |
| 2022-03-28 to 2022-04-01 | Sean |
| 2022-04-04 to 2022-04-08 | Dominykas |
| 2022-04-11 to 2022-04-15 | Ghislaine |
| 2022-04-18 to 2022-04-22 | Brent |
| 2022-04-25 to 2022-04-29 | Kriti |

### Instructions
You can also find the schedule and instructions for the Comms Assignee on the [wiki page](https://wiki.mathesar.org/team/guide/comms-assignee). Please feel free to update the wiki page if you think of anything useful during your shift.

**Reply by dmos62** [reply]
I'm on vacation from 04-01 to 04-12, which coincides with my assigned comms assignee window. @ghislaineguerin I see you're next in line after me. Would you mind switching places?

**Comment by seancolsen** [comment]
## Sean's check in

> How did your most recent work go?

- I spent almost the whole week working on the display and entry of numbers, which I thought would be relatively straightforward but turned out to have a lot of very interesting challenges, some of which are still unresolved. Although I'm a bit disappointed with with my slow progress, last week was one probably the most exciting weeks I've had on the job here so far because I've been learning some new stuff and working on a more challenging problem.

> What are you planning to work on soon?

- Unfortunately my planned list is basically the same as last week.
- Review/merge PRs from new contributors
- #970
- #930
- #1108
- #775


> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

- No

**Comment by dmos62** [comment]
>    How did your most recent work go?

While working on a niche feature (providing functionality for frontend to be able to provide "dropdown" value suggestions for equality-based filters), encountered some queries that service layer doesn't support well due to what seems to be concept friction. Currently, trying to land on a clear, synchronized outlook on these questions with Kriti. Relevant discussion: https://github.com/centerofci/mathesar/discussions/1154.

PS: I'll add that working on "highly different" features (like the one mentioned above) is fertile ground for optimizing Mathesar's architecture. I think it's a good idea to work on these as soon as possible. That way architectural changes will require less effort, and the simpler features will be subsequently easier to implement.

>    What are you planning to work on soon?

Get the above mentioned discussion sorted out. Get my submitted PR merged or iterate on it, depending on outcome of said discussion. Go on to work on tasks assigned to me, as seen in the backend priorities post.

>    Do you anticipate any dependencies or blockers that someone else on the team can help you with?

Diverse input on the mentioned dicussion (https://github.com/centerofci/mathesar/discussions/1154) would be interesting to hear.

**Comment by mathemancer** [comment]
> How did your most recent work go?

Last week was under-productive for a number of reasons. However, I'm glad to have clarified some issues surrounding locale. I'm also happy that the custom money type is finally merged (though some of the locale issues affect that as well).

> What are you planning to work on soon?

I'm working on grouping for number types currently. I'm also going to write some bugs about locale, and work on the date/time spec. I'm helping new contributors as well. After that, I'm jumping into more grouping.

> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

Not at the moment.

**Comment by kgodey** [comment]
> How did your most recent work go?

Pretty well, I got the Views product spec wrapped up, finally. I also did a whole bunch of comms around GSoC, including [a blog post](https://centerofci.org/2022/03/08/mathesar-participating-in-google-summer-of-code-2022/), social media posts, writing a lot of emails, triaging and assigning issues, helping new contributors etc. There was also a bunch of discussion last week.

> What are you planning to work on soon?

- Helping resolve #1154
- More GSoC comms
- #1008 

> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

Nope.

**Comment by pavish** [comment]
## Pavish's check in

> How did your most recent work go?

Last week was slow. Majority of types groundwork is done, but I have a lot of grunt work to complete: a range of e2e tests and some race conditions to fix. I wanted to create separate issues for the tests earlier but I've been noticing some regressions here and there on previously completed features and fixing them, and all of data types issues depend on the groundwork for which I didn't want to delay tests any further.

> What are you planning to work on soon?

My plan is almost the same as last week. Get the Text type updates merged and get the boolean type implementation complete, with groundwork for display options.

> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

None at the moment.

**Comment by ghislaineguerin** [comment]
> How did your most recent work go?

Last week was good, although with some context-switching due to temp brand work and additional reviews w Han. A lot of helpful insights from the conversations w Han.

Making progress on #1065 visual query builder design and thinking through the different scenarios. 

> What are you planning to work on soon?

#1065 Publish and review specs for visual query builder.

> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

- None

**Comment by silentninja** [comment]
> How did your most recent work go?

Last week was less productive than I expected. However, I am happy that most of the important discussions are coming to conclusion and which should speed things up.

> What are you planning to work on soon?

I am planning to work on the issues listed on the backend priorities followed by spending a day making some improvements for documentation and testing. 

> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

Not at the moment.
